infra/terraform/global/assets/prune_ebs_snapshots/snapshot/snapshot.go

```diff
@@ -0,0 +1,74 @@
+package snapshot
+
+import (
+	"../caller"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"log"
+	"time"
+)
+
+func GetSnapShots(connection *ec2.EC2, key string, val string) (output *ec2.DescribeSnapshotsOutput, err error) {
+
+	var snapShotInput *ec2.DescribeSnapshotsInput
+
+	// Get the owner ID a.k.a the aws account number
+	callerResult, callerErr := caller.GetCaller()
+	if callerErr != nil {
+		log.Panic(callerErr)
+	}
+	ownerID := *callerResult.Account
+
+	if len(key) != 0 && len(val) != 0 {
+		snapShotInput = &ec2.DescribeSnapshotsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String(fmt.Sprintf("tag:%s", key)),
+					Values: []*string{
+						aws.String(val),
+					},
+				},
+				{
+					Name: aws.String("owner-id"),
+					Values: []*string{
+						aws.String(ownerID),
+					},
+				},
+			},
+		}
+	} else {
+		snapShotInput = &ec2.DescribeSnapshotsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("owner-id"),
+					Values: []*string{
+						aws.String(ownerID),
+					},
+				},
+			},
+		}
+	}
+
+	output, err = connection.DescribeSnapshots(snapShotInput)
+
+	return output, err
+}
+
+func PruneSnapShots(connection *ec2.EC2, snapshotID string) (output *ec2.DeleteSnapshotOutput, err error) {
+
+	deleteSnapShotShotInput := &ec2.DeleteSnapshotInput{
+		SnapshotId: aws.String(snapshotID),
+	}
+
+	output, err = connection.DeleteSnapshot(deleteSnapShotShotInput)
+
+	return output, err
+}
+
+func CheckSnapShotAge(SnapShotStartTimeStamp *time.Time, duration time.Duration) (isOlder bool, err error) {
```

Comment:
You could rename few things here if you feel like it, e.g.
- This function could be renamed to `OlderThan()` as "snapshot" is implicit as you're in the `snapshots` package
- `SnapShotStartTimeStamp` => `createdAt` or something like that

Probably this could be:

```go
func OlderThan(createdAt *time.Time, duration time.Duration) bool {
    return createdAt.Add(duration).Before(time.Now())
}
```

Not a massive difference really, just a bit easier on the eyes maybe. 
Also, you may argue that this function is not really snapshots-specific so it may live somewhere else. I'd not worry too much about that tho, just something to keep in mind in case in the future you see the opportunity to factor this out.

Reply:
👍